src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,842 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+            } else {
+                dmg = monSpAtk;
+            }
+        } else {
+            dmg = monSpAtk * sStatChangeTable[statChangeSpAtk][0] / sStatChangeTable[statChangeSpAtk][1];    
+        }
+        
+        dmg *= movePower;
+        dmg *= ((level * 2 / 5) + 2);
+        
+        if (crit > 1) {
+            if (statChangeSpDef < 6) {
+                dmg2 = monSpDef * sStatChangeTable[statChangeSpDef][0] / sStatChangeTable[statChangeSpDef][1];
+            } else {
+                dmg2 = monSpDef;
+            }
+        } else {
+            dmg2 = monSpDef * sStatChangeTable[statChangeSpDef][0] / sStatChangeTable[statChangeSpDef][1];    
+        }
+        
+        dmg /= dmg2;
+        dmg /= 50;
+        
+        if ((sideCondition & SIDE_CONDITION_LIGHT_SCREEN) && crit == 1 && ctx->unk_334.moveData[moveNo].effect != 186) {
+            if ((battleType & BATTLE_TYPE_DOUBLES) && GetMonsHitCount(bsys, ctx, 1, battlerIdTarget) == 2) {
+                dmg = dmg * 2 / 3;
+            } else {
+                dmg /= 2;
+            }
+        }
+    }
+    
+    if ((battleType & BATTLE_TYPE_DOUBLES) && ctx->unk_334.moveData[moveNo].range == 4 && GetMonsHitCount(bsys, ctx, 1, battlerIdTarget) == 2) {
+        dmg = dmg * 3 / 4;
+    }
+    if ((battleType & BATTLE_TYPE_DOUBLES) && ctx->unk_334.moveData[moveNo].range == 8 && GetMonsHitCount(bsys, ctx, 0, battlerIdTarget) >= 2) {
+        dmg = dmg * 3 / 4;
+    }
+    
+    if (!CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_CLOUD_NINE) && !CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_AIR_LOCK)) {
+        if (fieldCondition & FIELD_CONDITION_RAIN_ALL) {
+            switch (moveType) {
+            case TYPE_FIRE:
+                dmg /= 2;
+                break;
+            case TYPE_WATER:
+                dmg = dmg * 15 / 10;
+                break;
+            }
+        }
+        
+        if ((fieldCondition & FIELD_CONDITION_WEATHER_NO_SUN) && moveNo == MOVE_SOLAR_BEAM) {
+            dmg /= 2;
+        }
+        if (fieldCondition & FIELD_CONDITION_SUN_ALL) {
+            switch(moveType) {
+            case TYPE_FIRE:
+                dmg = dmg * 15 / 10;
+                break;
+            case TYPE_WATER:
+                dmg /= 2;
+                break;
+            }
+        }
+    }
+    
+    if (GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_FLASH_FIRE_ACTIVE, NULL) && moveType == TYPE_FIRE) {
+        dmg = dmg * 15 / 10;
+    }
+    
+    return dmg + 2;
+}
+
+int ApplyDamageRange(BattleSystem *bsys, BATTLECONTEXT *ctx, int damage) {
+    if (damage) {
+        damage *= (100 - (BattleSystem_Random(bsys) % 16));
+        damage /= 100;
+        if (!damage) {
+            damage = 1;
+        }
+    }
+    return damage;
+}
+
+extern u8 sCritChance[5];
+
+u32 TryCriticalHit(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerIdAttacker, int battlerIdTarget, int critCnt, u32 sideCondition) {
```

Comment:
name seems to imply it tries to perfom a critical hit, where it just calculates the chance for it

Reply:
The return value is the multiplier for the critical hit, so 1 if it doesn't work and 2-3 if it does (depending on if the mon has sniper)

Reply:
I'd probably prefer something like CalculateCriticalHit

Reply:
Eh? I prefer Try but they functionally mean the same thing, so just gonna leave it as is